bundles/framework/admin-announcements/view/AnnouncementsForm.jsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import "antd/dist/antd.css";
+import PropTypes from 'prop-types';
+import { Form, Input, Button, DatePicker, Switch, Row } from "antd";
+import { Message, Confirm } from 'oskari-ui';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import moment from 'moment';
+
+/*
+This file contains the form for admin-announcements.
+This is the main file for creating and editing announcements.
+*/
+
+const { RangePicker } = DatePicker;
+const { TextArea } = Input;
+const rangeConfig = {
+  rules: [
+    {
+      type: "array",
+      required: true,
+      message: <Message messageKey='dateError' />
+    }
+  ]
+};
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='admin-announcements' />;
+
+const dateFormat = 'YYYY-MM-DD';
+
+const AnnouncementsForm = ({controller,  title, key, form, index}) => {
+
+  const onFinish  = fieldsValue => {
+    // Should format date value before submit.
+    const rangeValue = fieldsValue["range_picker"];
+
+    //If creating new announcement
+    if (form.id === undefined) {
+      const values = {
+        title: fieldsValue["title"],
+        content: fieldsValue["content"],
+        begin_date: rangeValue[0].format('YYYY-MM-DD'), 
+        end_date: rangeValue[1].format('YYYY-MM-DD'),
+        active: fieldsValue["active"]
+      };
+      controller.saveAnnouncement(values);
+      
+    } 
+
+    // Else when editing old announcement
+    else {
+
+      const values = {
+        id: form.id,
+        title: fieldsValue["title"],
+        content: fieldsValue["content"],
+        begin_date: rangeValue[0].format('YYYY-MM-DD'), 
+        end_date: rangeValue[1].format('YYYY-MM-DD'),
+        active: fieldsValue["active"]
+      };
```

Comment:
Code is duplicated here. You can do a common part for the values outside the if/else and just add id on edit + the method calls inside if/else.

Reply:
refactored